answers.txt

```diff
@@ -1,19 +1,19 @@
 1. What is a gitignore file and what should it include for a Java project?
 
-A. gitingnore file consists of lists of extensions which the repository should not contain.
-   For example, for a java project we do not want .class, .project or .jar and .war files to be included in our repoitory. We would need only .java files mainly for our source control.
-   In that case, the gitignore file will have .calss, .project, .jar etc as its contents
+A. gitingnore file consists of lists of extensions which the repository should not contain
+   For example, for a java project we do not want .class, .project or .jar and .war files to be included in our repoitory. We would need only .java files mainly for our source control
```

Comment:
Java projects create various kinds of extension based on the IDE as well. Can you please name some files which are to be ignored for Eclipse and IntelliJ?

Reply:
For IntelliJ, below file extensions can be ignored
.idea/modules.xml
.idea/*.iml
.idea/modules
*.iml
*.ipr

For Eclipse, below file extensions can be ignored
.settings/
.classpath
.project